config/puma/production.rb

```diff
@@ -27,6 +27,22 @@
 #
 preload_app!
 
+# Check if FOREMAN_BIND was set to an IP address based on the previous
+# definition of FOREMAN_BIND. If it is an IP address, define the host
+# and port through Puma's DSL. Otherwise rescue and assume the new
+# style of a fully specified bind in the formats of:
+#
+#  * unix:///run/foreman.sock
+#  * tcp://127.0.0.1:3000
+#
+begin
+  IPAddr.new(ENV['FOREMAN_BIND'])
+
+  port ENV.fetch('FOREMAN_PORT', '3000'), ENV['FOREMAN_BIND']
+rescue
```

Comment:
I'm never very happy with rescue without a specific exception. Now if `port` fails, it'll also raise an exception that's caught here.
```suggestion
rescue IPAddr::Error
```